lib/Model.php

```diff
@@ -1751,7 +1774,14 @@ protected static function Relation(array $options = []): Relation
      */
     public static function count(/* ... */): int
     {
-        return static::Relation()->count(...func_get_args());
+        $relation = static::Relation();
```

Comment:
Backwards compatibility for version 1. What I had in my original PR was `Model::find()` would have the complex support for version 1 parameters, and `Relation::find()` would only support what's in modern Ruby. If you want to break every single user of version 1 (ie: all of us) in version 2, then this is no longer necessary. The users of version 2 would do `Model::where(complex input stuff)->count()`

Reply:
if someone wants to start a 1.x legacy branch and start back-porting fixes then that's up to them, but otherwise I'm not interested in bc; this API is confusing enough without muddying things further. Once we have our shiny new 2.x line, we will be more careful going forward.

Reply:
Oh, so that's why you removed all those version 1 tests and threw away the code I wrote in my original PR to be version 1 compatible. If that's not a design goal, then that simplifies implementation, but will probably upset the entire version 1 user base as everybody will need to change their code. But that's the meaning of going from version 1 to 2. It was possible to keep backwards compatibility like I showed for count(). Anyway, removed backwards compatibility.

Reply:
Yeah, sorry about that, but even if we did keep passing where clauses to `find` and `count` it still wouldn't make v2.0 compatible with 1.x (because, for example, I moved us away from mixed-array association configurations in the interest of static checking). I guess what we could do is find the earliest point in the commit history where all the tests pass and tag it with 1.2, or something, but we would have to revert the config changes, and I just don't have the energy. Feel free to volunteer, though.

Reply:
We could look at cc365079d15c8e758b8cd4e06b34e7c4d9be5625 as a potential 1.2 release. I'll try it out in my own codebase at some point and see what happens.